src/components/App.test.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import { render, screen, waitForElementToBeRemoved } from '@testing-library/react'
+import { act } from 'react-dom/test-utils'
+import user from '@testing-library/user-event'
+
+import { BrowserRouter } from 'react-router-dom'
+
+import { SettingsProvider } from '../context/SettingsContext'
+import { WalletProvider } from '../context/WalletContext'
+import { WebsocketProvider } from '../context/WebsocketContext'
+
+import App from './App'
+
+const AllTheProviders = ({ children }) => {
+  return (
+    <BrowserRouter>
+      <SettingsProvider>
+        <WalletProvider>
+          <WebsocketProvider>{children}</WebsocketProvider>
+        </WalletProvider>
+      </SettingsProvider>
+    </BrowserRouter>
+  )
+}
```

Comment:
Should we make this reusable so that we don't have to update all the test files when we change something in the providers structure?

Reply:
Jep. Moved it to dir `__utils__` to hopfully let everyone know by convention that it is intended to be used in tests. Would have wanted it to be part of the `global.__DEV__` variable, but unfortunately some Providers need `window.JM` already present at import. Tried to refactor it a little bit but makes things far more complicated at first glance. It might be annoying to import utils in every file, but then again, not all components need all providers.

Hope this is acceptable - suggestions welcome.